qstash/workflow/quickstarts/cloudflare-workers.mdx

```diff
@@ -0,0 +1,167 @@
+---
+title: "Cloudflare Workers"
+---
+
+This guide provides detailed, step-by-step instructions on how to use and deploy Upstash Workflows with Cloudflare Workers. You can also explore our [Cloudflare Workers example](https://github.com/upstash/qstash-js/tree/main/examples/workflow/cloudflare-workers) or [Hono.js Cloudflare Workers example](https://github.com/upstash/qstash-js/tree/main/examples/workflow/cloudflare-workers-hono) for detailed, end-to-end examples and best practices.
+
+## Prerequisites
+
+1. An Upstash QStash API key.
+2. Node.js and npm (another package manager) installed.
+
+If you haven't obtained your QStash API key yet, you can do so by [signing up](https://console.upstash.com/login) for an Upstash account and navigating to your QStash dashboard.
+
+## Step 1: Installation
+
+First, install the QStash SDK in your worker project:
+
+```bash Terminal
+npm install @upstash/qstash
+```
+
+## Step 2: Configure Environment Variables
+
+Create a `.dev.vars` file in your project root and add your QStash token. This key is used to authenticate your application with the QStash service.
+
+```bash Terminal
+touch .dev.vars
+```
+
+Open the environment file and add your QStash token, as well as the environment you're running in:
+
+```bash .dev.vars
+QSTASH_TOKEN=xxxxxxxxx
+ENVIRONMENT=development
+```
+
+Replace `xxxxxxxxx` with your actual QStash token found here:
+
+<Frame>
+  <img src="/img/qstash-workflow/qstash_token.png" />
+</Frame>
+
+## Step 3: Create a Workflow Endpoint
+
+A workflow endpoint allows you to define a set of steps that, together, make up a workflow. Each step contains a piece of business logic that is automatically retried on failure, with easy monitoring via our visual workflow dashboard.
+
+To define a workflow endpoint with Cloudflare Workers, navigate into your workers entrypoint file (usually `src/index.ts`) and add the following code:
+
+```typescript src/index.ts
+import { serve } from "@upstash/qstash/cloudflare"
+
+interface Env {
+  ENVIRONMENT: "development" | "production"
+}
+
+export default {
+  fetch(req: Request, env: Env, ctx: ExecutionContext) {
+    // 👇 For TypeScript users: define initial payload type
+    return serve<{ text: string }>(
+      async (context) => {
```

Comment:
this example doesn't look correct. note that we don't do anything with req/env/ctx variables.

It should have been like [this](https://github.com/upstash/docs/pull/238/commits/5af4011063a0270cd4153392d221335bdfc68013#diff-f0bab82153766e46e18e860e3acdb9486170693c3a9f658cd6da83b1c3bc95d1R65)

Reply:
this example comes straight from the CF workers boilerplate, what about it is not correct?

Reply:
[serve method of cloudflare handles the req/env/ctx parameters in it](https://github.com/upstash/qstash-js/blob/main/platforms/cloudflare.ts#L64). This way, you can basically say and let serve do the work:
```javascript
export default {
  fetch: serve<{text: string}>(async context => {
    // ...
  }
}
```

see our example [here](https://github.com/upstash/qstash-js/blob/main/examples/workflow/cloudflare-workers/src/index.ts)

Reply:
this issue remains in three cf examples @joschan21